apps/functions-subscription/src/adapters/azure/functions/__tests__/list-trials.test.ts

```diff
@@ -0,0 +1,124 @@
+import { describe, expect, it } from 'vitest';
+import * as TE from 'fp-ts/TaskEither';
+import { makeAValidListTrialRequest, supportHttpRequestHeaders } from './data';
+import { makeFunctionContext, makeTestSystemEnv } from './mocks';
+import { makeListTrialsHandler } from '../list-trials';
+import { aTrial } from '../../../../domain/__tests__/data';
+import { HttpRequest } from '@azure/functions';
+import { Trial, TrialId } from '../../../../domain/trial';
+import { NonEmptyString } from '@pagopa/ts-commons/lib/strings';
+
+describe('makeListTrialsHandler', () => {
+  const aSlimTrial = {
+    id: aTrial.id,
+    name: aTrial.name,
+    state: aTrial.state,
+    description: aTrial.description,
+  } as Trial;
+
+  const anotherSlimTrial = {
+    ...aSlimTrial,
+    id: 'anotherTrialId' as TrialId,
+    name: 'anotherTrialName' as NonEmptyString,
+  } as Trial;
```

Comment:
**issue**: slim trial is a concept introduced into the HTTP layer (coming from OpenAPI).

if you cast to `Trial`, then I assume the objects are not `SlimTrial`.

why not using the `aTrial` object already available into the `data.ts` file?

Reply:
when using `aTrial`, the return object is different from the former as it get stripped of the author property, making the test fail (`AssertionError: expected { items: [ { …(4) }, { …(4) } ], …(2) } to match object { items: [ { …(5) }, { …(5) } ], …(2) }`).
i think it is correct to expect a TrialSlim object, maybe a solution would be to update `toTrialListAPI` to use a `toSlimTrialAPI` instead of `toTrialAPI`?

Reply:
reading the following comment i think i get what you meant now. i'll update it following your suggestion.

Reply:
Updated in 3dc0bb86816f2a2f2f3fbcf3807f68fc3e45837a